tests/test_assembly.py

```diff
@@ -218,3 +218,10 @@ def test_empty_solve(nested_assy):
 
     with pytest.raises(ValueError):
         nested_assy.solve()
+
+
+def test_expression_grammar(nested_assy):
+
+    nested_assy.constrain(
+        "TOP@faces@>Z", "SECOND/BOTTOM@vertices@>X and >Y and >Z", "Plane"
```

Comment:
That's going to error if you try to solve it (or if we implement any kind of constrain validation at a later date). Any reason not to use `"Point"`?

```suggestion
        "TOP@faces@>Z", "SECOND/BOTTOM@vertices@>X and >Y and >Z", "Point"
```

Reply:
No reason, I just wanted to test the selectors. Indeed it is good to keep the code correct.